2D_platformer_template/platformer_stock.py

```diff
@@ -0,0 +1,897 @@
+import pygame
+import random
+import math
+
+DARK_RED = (139, 0, 0)
+YELLOW = (235, 195, 65)
+BLACK = (0, 0, 0)
+ORANGE = (255, 165, 0)
+BLUE = (30, 144, 255)
+CYAN = (47, 237, 237)
+RED = (194, 57, 33)
+LIME_GREEN = (50, 205, 50)
+LIGHT_RED = (255, 0, 0)
+WHITE = (255, 255, 255)
+GREY = (125, 125, 125)
+LIGHT_PINK = (255, 182, 193)
+DARK_GREEN = (1, 100, 32)
+PURPLE = (181, 60, 177)
+BROWN = (150, 75, 0)
+DARK_GREY = (52, 52, 52)
+
+
+class Text:
+    """
+    Class used to simplify text creation for pygame
+    """
+
+    def __init__(self, text, text_pos, font_size, font_type,
+                 font_color, text_other):
+        self.text = text  # Text as a string
+        self.position = text_pos  # Text position as a tuple or list (x and y)
+        self.font_size = int(font_size)  # Int determining how big the text is
+        self.font_type = font_type  # String used to indicate what font
+        """Font selection is determined by your computer and it's preset fonts
+        """
+        self.color = font_color
+        """A constant string for a tuple or a tuple using RGB values"""
+        self.other = text_other
+        """PLACEHOLDER for any other variables needed or desired in text"""
+        self.font = None  # Initialized here, defined in setup()
+        self.text_rect = None  # Initialized here, defined in render()
+        self.text_img = None  # Initialized here, defined in render()
+
+        self.setup()  # Called to set up the font
+        self.render()
+        """Called to continuously update the position, rect, color, and text
+        """
+
+    def setup(self):
+        """
+        Uses font type and size to translate into pygame text font
+        and to make self.font
+        """
+        self.font = pygame.font.SysFont(self.font_type, self.font_size)
+
+    def render(self):
+        """
+        Creates self.text_img or the pygame image of the text using self.text,
+            self.color.
+        Creates self.text_rect, or a rect object using the size of the text.
+        Then centers the rect around the text (or the defined position)
+        """
+        self.text_img = self.font.render(self.text, True, self.color)
+        self.text_rect = self.text_img.get_rect()
+        self.text_rect.center = self.position
+
+    def scale(self, width, height):
+        self.position = list(self.position)
+        self.position[0] = int(self.position[0] * width)
+        self.position[1] = int(self.position[1] * height)
+        self.position = tuple(self.position)
+        self.font_size = int(self.font_size * max(width, height))
+
+        # Apply those changes
+        self.setup()
+        self.render()
+
+
+class Music:

```

Comment:
The template used was a previous version of the base game stock, meaning that music compatibility and checks aren't present